### PR TITLE
status quo: Alan tries processing some files

### PR DESCRIPTION
**Pull request by mypetyak**
I recently found https://blog.rust-lang.org/2021/03/18/async-vision-doc.html, and thought I'd contribute my experience as someone learning new to `async` Rust (and Rust in general).

### CODE REVIEW

File: src/vision/status_quo/alan_tries_processing_some_files.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Alan is new to Rust. He wants to build a program that recurses over all the files in a directory (and its subdirectories), reads each file, and produces some fingerprint of the file.
+
+Since so much blocking I/O is involved, he chooses async in order to process many files concurrently.
+
+### Recursion
+Alan starts by writing a recursive function that can call some operation on each regular file in a directory and recurse on each subdirectory.
```
Comment (nikomatsakis) [inline src/vision/status_quo/alan_tries_processing_some_files.md:16]: One thing I wondered here: is Alan using tokio? How did he decide to use tokio?

File: src/vision/status_quo/alan_tries_processing_some_files.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+### **What are the morals of the story?**
+- The manual desugaring required for `async` recursion erases some of the "magic" of `async`.  
+- Some programmers may never implement custom types that are `Future`, instead using standard constructs like `async` blocks to produce them. In these cases, the programmer might assume the returned `Future`s should have concrete types with known sizes, which would allow them to work directly with the returned types rather than have to deal with the complexities of trait objects, `Box`-ing, and opaque type comparisons.
+- `Pin` documentation focuses on data that can or cannot "move" in memory. To someone new to Rust, it might be easy to confuse this concept with "move" semantics in the context of ownership.
+
+### **What are the sources for this story?**
+I describe my own experience while working on my first Rust project.
+
+### **Why did you choose *NAME* to tell this story?**
```
Comment (nikomatsakis) [inline src/vision/status_quo/alan_tries_processing_some_files.md:272]:
```suggestion
### **Why did you choose Alan to tell this story?**
```

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
This is really great! Thanks. I had one question and one minor suggestion.

File: src/vision/status_quo/alan_tries_processing_some_files.md (hunk 1 above)
Comment (mypetyak) [reply src/vision/status_quo/alan_tries_processing_some_files.md:27]: Ah, good point - I've added an `### Async` section to provide a little bit about ~~my~~ Alan's route to using Tokio.

**Review by eminence — APPROVED** [review]

**Pull request merged**